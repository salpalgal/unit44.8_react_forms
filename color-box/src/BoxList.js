import React ,{useState} from "react";
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
import {v4 as uuid} from "uuid"



const BoxList = ()=>{
    
    
    const [boxes ,setBoxes]= useState([]);
    const addBox = (newBox)=>{
        setBoxes(boxes=>[...boxes,{...newBox, id:uuid()}])
    }
    const RemoveBox = id=>{
        setBoxes(boxes => boxes.filter(box=> box.id !== id))
    }
    const allBoxs = boxes.map(({width,height,color,id})=> 
    <Box width={width} height={height} color={color} id={id} handleRemove={RemoveBox}/>)
    

return (
    <div>
        <NewBoxForm addBox={addBox} />
        {allBoxs}
        
            
        {/* // <Box width={100} height={100} color="darkgreen" /> */}
    </div>
)
}

export default BoxList;