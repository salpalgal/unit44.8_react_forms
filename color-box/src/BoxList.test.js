import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxlist, height="5",width="5",color="blue"){
    const heightInput = boxlist.getByLabelText("Height")
    const widthInput = boxlist.getByLabelText("Width")
    const colorInput = boxlist.getByLabelText("Color")
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    fireEvent.change(colorInput, { target: { value: color } });
    const button = boxlist.getByText("Sumbit")
    fireEvent.click(button)
}
it("works without crashing", function(){
  render(<BoxList />)
})

it("can add box", function(){
    const boxlist = render(<BoxList/>)

    expect(boxlist.queryByText("X")).not.toBeInTheDocument();
    addBox(boxlist)
    
    const removeBtn = boxlist.queryByText("X")
    expect(removeBtn).toBeInTheDocument(); 
    
})
it("can remove box", function(){
    const boxlist = render(<BoxList />)
    addBox(boxlist)

    const removeBtn = boxlist.getByText("X");

    fireEvent.click(removeBtn)

    expect(removeBtn).not.toBeInTheDocument()
})

