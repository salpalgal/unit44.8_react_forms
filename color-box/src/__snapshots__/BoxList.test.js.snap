// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
         Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
         Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Enter height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
         Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="Enter color"
        type="text"
        value=""
      />
      <button>
        Sumbit
      </button>
    </form>
    <div>
      <div />
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
